ingest-app/src/cmr/ingest/validation/generic_document_validation.clj

```diff
@@ -0,0 +1,166 @@
+(ns cmr.ingest.validation.generic-document-validation
+  "Provides functions to validate the ingest generic document"
+  (:require
+   [cheshire.core :as json]
+   [clojure.string :as string]
+   [cmr.common.cache :as cache]
+   [cmr.common.cache.in-memory-cache :as mem-cache]
+   [cmr.common.concepts :as concepts]
+   [cmr.common.generics :as generics]
+   [cmr.common.log :refer (info error)]
+   [cmr.common.services.errors :as errors]
+   [cmr.transmit.metadata-db :as mdb]))
+
+(def schema-validation-cache-key
+  "The cache key for the schema validation functions cache."
+  :schema-validation-functions)
+
+(def SCHEMA_CACHE_TIME
+  "The number of milliseconds schema validation functions will be cached."
+  (* 24 60 60 1000)) ;; 24 hours
+
+(defn create-schema-validation-cache
+  "Creates a cache for schema validation functions."
+  []
+  (mem-cache/create-in-memory-cache :ttl {} {:ttl SCHEMA_CACHE_TIME}))
```

Comment:
why not redis? Could memoize also work? Just wondering what lead to doing only the in-memory cache

Reply:
Redis latency instead of file io seems like an equal exchange, since we are just trying to avoid reading the file over and over a memoization could work instead of in memory, but with in memory we can set a ttl which I find more comfortable.  But I am not married to any solution.